### PR TITLE
Added query parameters for lat/lon in service and tests, need to fix …

### PR DESCRIPTION
**Pull request by MC-Student**
…in MVC. Added TODO notes for displaying the information and using the user's input for lat/lon

### CODE REVIEW

File: src/main/java/cohen/FreebieService.java [hunk 1]
```diff
@@ -3,12 +3,14 @@
...
 public interface FreebieService
 {
-    @GET("api/v1.3/posts?sort_by=date&types=offer&sources=open_archive_groups&" +
-            "per_page=20&page=1&device_pixel_ratio=1&latitude=40.776676&" +
-            "longitude=-73.971321&radius=21000&date_min=2023-05-01T18%3A30%3A16&" +
-            "include_reposts=0&api_key=Q7gR3Ez9sBuwPS7OEMuHy6rP1TFWP29ylI5wghNC")
-    Observable<PostListInfo> getPostList();
+    @GET("api/v1.3/posts?sort_by=date&types=offer&sources=open_archive_groups&"
+            + "per_page=20&page=1&device_pixel_ratio=1&radius=21000&"
+            + "date_min=2023-05-01T18%3A30%3A16&include_reposts=0&"
```
Comment (andrewoid) [inline src/main/java/cohen/FreebieService.java]: `date_min` should be a parameter.

File: src/main/java/cohen/dagger/FreebieServiceModule.java [hunk 2]
```diff
@@ -27,10 +29,25 @@ public FreebieService providesFreebieService()
...
+    public JList<String> providesPostTitles()
     {
-        return new JTextArea();
+        return new JList<>();
     }
+
+    @Provides
+    @Named("allPosts")
+    @Singleton
+    public ArrayList<Post> providesAllPosts(){return new ArrayList<>();}
```
Comment (andrewoid) [inline src/main/java/cohen/dagger/FreebieServiceModule.java]: reformat your code.

File: src/main/java/cohen/mvc/FreebiePostsController.java [hunk 3]
```diff
@@ -3,41 +3,67 @@
...
     }
 
-    public void setPostList(PostListInfo postListInfo)
+    public void setPosts(PostListInfo postListInfo)
     {
-        List<Post> posts = postListInfo.getPosts();
-        for (Post post : posts)
+        allPosts = (ArrayList<Post>) postListInfo.getPosts();
+
+        postTitles.setListData((String[]) null);
```
Comment (andrewoid) [inline src/main/java/cohen/mvc/FreebiePostsController.java]: Why are you doing this?

File: src/main/java/cohen/mvc/FreebiePostsFrame.java [hunk 4]
```diff
@@ -1,40 +1,166 @@
...
+            public void keyPressed(KeyEvent e)
+            {
+            }
+
+            @Override
+            public void keyReleased(KeyEvent e)
+            {
+            }
+        });
+        userLat.setFocusable(true);
```
Comment (andrewoid) [inline src/main/java/cohen/mvc/FreebiePostsFrame.java]: This should automatically be focusable.

File: src/main/java/cohen/mvc/FreebiePostsFrame.java [hunk 5]
```diff
@@ -1,40 +1,166 @@
...
+            }
+
+            @Override
+            public void keyReleased(KeyEvent e)
+            {
+
+            }
+        });
+        userLon.setFocusable(true);
+        userLon.requestFocus();
```
Comment (andrewoid) [inline src/main/java/cohen/mvc/FreebiePostsFrame.java]: This should be focusable already.

Two things can't both request focus.

File: src/main/java/cohen/mvc/FreebiePostsFrame.java [hunk 6]
```diff
@@ -39,7 +32,7 @@ public FreebiePostsFrame(FreebiePostsController controller,
         dateTime.setHorizontalAlignment(SwingConstants.CENTER);
         DateFormat df = new SimpleDateFormat("yyyy-MM-dd");
         userDate = new JFormattedTextField(df);
-        userDate.setText("2023-05-17");
+        userDate.setText("2023-05-24");
```
Comment (andrewoid) [inline src/main/java/cohen/mvc/FreebiePostsFrame.java]: I meant, programmatically get today's date.

File: src/main/java/cohen/mvc/FreebiePostsFrame.java [hunk 7]
```diff
@@ -188,19 +189,21 @@ public void valueChanged(ListSelectionEvent e)
...
-        controller.refreshPosts(userLat.getText(), userLon.getText(), getUTCDateParam());
+        controller.refreshPosts(userLat.getText(), userLon.getText(), getUtcDateParam());
 
         setContentPane(mainPanel);
     }
 
-    private String getUTCDateParam()
+    private String getUtcDateParam()
     {
         return userDate.getText() + "T00%3A00%3A00";
```
Comment (andrewoid) [inline src/main/java/cohen/mvc/FreebiePostsFrame.java]: The `%3A` is the ":" character, just escaped. You should be using ":" instead.
https://agileappscloud.info/aawiki/UTC_Format

Comment (MC-Student) [reply src/main/java/cohen/mvc/FreebiePostsFrame.java]: Right, thank you. Thought it needed to be escaped but that would make more sense, and is probably why the service was having issues

File: src/main/java/cohen/mvc/FreebiePostsController.java [hunk 8]
```diff
@@ -61,13 +69,51 @@ public void setPosts(PostListInfo postListInfo)
...
+        photo.setIcon(setPhoto(postSelected));
+    }
+
+    public ImageIcon setPhoto(int index) throws MalformedURLException
+    {
+        String url = allPosts.get(index).getPhotos().get(index).getUrl();
+        ImageIcon icon = new ImageIcon(new URL(url));
+        Image origImage = icon.getImage();
+        Image scaledInstance = origImage.getScaledInstance(600, 800, java.awt.Image.SCALE_SMOOTH);
+        return new ImageIcon(scaledInstance); //this returns null because does not wait for image to load
```
Comment (andrewoid) [inline src/main/java/cohen/mvc/FreebiePostsController.java]: This comment is out of date.

File: src/main/java/cohen/mvc/FreebiePostsController.java [hunk 9]
```diff
@@ -61,13 +69,51 @@ public void setPosts(PostListInfo postListInfo)
...
+        /*
+
+private static boolean openWebpage(URL url) {
```
Comment (andrewoid) [inline src/main/java/cohen/mvc/FreebiePostsController.java]: Why is this code commented out?

File: src/main/java/cohen/mvc/FreebiePostsFrame.java [hunk 10]
```diff
@@ -22,7 +28,8 @@ public class FreebiePostsFrame extends JFrame
     public FreebiePostsFrame(FreebiePostsController controller,
                              @Named("postTitles") JList<String> postTitles,
```
Comment (andrewoid) [inline src/main/java/cohen/mvc/FreebiePostsFrame.java:32]: Why is this a constructor parameter?

File: src/main/java/cohen/mvc/FreebiePostsController.java [hunk 11]
```diff
@@ -3,41 +3,111 @@
...
+            String[] titlesArray = new String[allPosts.size()];
+
+            for (int i = 0; i < titlesArray.length; i++)
+            {
+                titlesArray[i] = allPosts.get(i).getTitle();
+            }
+
+            postTitles.setListData(titlesArray);
+
+            title.setText(allPosts.get(0).getTitle());
```
Comment (andrewoid) [inline src/main/java/cohen/mvc/FreebiePostsController.java]: call `allPosts.get(0)` only once.

File: src/main/java/cohen/mvc/FreebiePostsFrame.java [hunk 12]
```diff
@@ -3,38 +3,305 @@
...
+
+    private String getUtcDateParam()
+    {
+        return userDate.getText() + "T00:00:00";
+    }
+
+    private boolean allInputComplete()
+    {
+        return !userLat.getText().isBlank()
+                & !userLon.getText().isBlank()
```
Comment (andrewoid) [inline src/main/java/cohen/mvc/FreebiePostsFrame.java]: use &&

File: src/main/java/cohen/mvc/FreebiePostsController.java (hunk 8 above)
Comment (MC-Student) [reply src/main/java/cohen/mvc/FreebiePostsController.java]: This is an old section of code that is outdated...

File: src/main/java/cohen/mvc/FreebiePostsController.java (hunk 9 above)
Comment (MC-Student) [reply src/main/java/cohen/mvc/FreebiePostsController.java]: Again, this is an old version.

File: src/main/java/cohen/dagger/FreebieServiceModule.java [hunk 13]
```diff
@@ -27,10 +29,42 @@ public FreebieService providesFreebieService()
...
+        return new JList<>();
+    }
+
+    @Provides
+    @Named("allPosts")
+    @Singleton
+    public ArrayList<Post> providesAllPosts()
+    {
+        return new ArrayList<>();
+    }
```
Comment (andrewoid) [inline src/main/java/cohen/dagger/FreebieServiceModule.java]: This is unused

File: src/main/java/cohen/mvc/FreebiePostsFrame.java (hunk 10 above)
Comment (MC-Student) [reply src/main/java/cohen/mvc/FreebiePostsFrame.java:32]: So that the menu of titles on the side can be easily updated whenever the Observable<> changes

Comment (MC-Student) [reply src/main/java/cohen/mvc/FreebiePostsFrame.java:32]: Do I need to change this? I tried changing it to just be a method called from the controller when anything gets updated, but I am now having an error where I cannot determine the cause, but I know it crashes on the first line where this method is called.

**Comment by MC-Student** [comment]
The changes I made before due to the list action listener were erased, which I forgot to take into consideration. How did I fix that issue? Instead of calling !e.getValueIsAdjusting()

**Review by andrewoid — APPROVED** [review]

**Pull request merged**